streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -133,9 +132,9 @@ public StateStore getStateStore(String name) {
         if (node == null)
             throw new TopologyBuilderException("Accessing from an unknown node");
 
-        // TODO: restore this once we fix the ValueGetter initialization issue
-        //if (!node.stateStores.contains(name))
-        //    throw new TopologyBuilderException("Processor " + node.name() + " has no access to StateStore " + name);
+        if (!node.stateStores.contains(name)) {
```

Comment:
Will add a test.
About the TODO (it is unclear to Guozhang and me what the issues was, and if it got resolved or not). Because all tests pass, we assume it's fine. Need to keep it in mind though, if we run into a new (or than maybe old) issue...